src/js/component.js

```diff
@@ -82,1182 +90,1103 @@ var Component = CoreObject.extend({
- *     var myComp = new MyComponent(player, {
- *       children: {
- *         myChildComponent: {
- *           myChildOption: true
- *         }
- *       }
- *     });
- *
- * The children option can also be an Array of child names or
- * child options objects (that also include a 'name' key).
- *
- *     var myComp = new MyComponent(player, {
- *       children: [
- *         'button',
- *         {
- *           name: 'button',
- *           someOtherOption: true
- *         }
- *       ]
- *     });
- *
- */
-Component.prototype.initChildren = function(){
-  let parent = this;
-  let parentOptions = parent.options();
-  let children = parentOptions['children'];
-
-  let handleAdd;
-  if (children) {
-    handleAdd = function(name, opts){
-      // Allow options for children to be set at the parent options
-      // e.g. videojs(id, { controlBar: false });
-      // instead of videojs(id, { children: { controlBar: false });
-      if (parentOptions[name] !== undefined) {
-        opts = parentOptions[name];
-      }
 
-      // Allow for disabling default components
-      // e.g. vjs.options['children']['posterImage'] = false
-      if (opts === false) return;
-
-      // Create and add the child component.
-      // Add a direct reference to the child by name on the parent instance.
-      // If two of the same component are used, different names should be supplied
-      // for each
-      parent[name] = parent.addChild(name, opts);
-    };
+  /**
+   * Add an event listener to this component's element
+   *
+   *     var myFunc = function(){
+   *       var myComponent = this;
+   *       // Do something when the event is fired
+   *     };
+   *
+   *     myComponent.on('eventType', myFunc);
+   *
+   * The context of myFunc will be myComponent unless previously bound.
+   *
+   * Alternatively, you can add a listener to another element or component.
+   *
+   *     myComponent.on(otherElement, 'eventName', myFunc);
+   *     myComponent.on(otherComponent, 'eventName', myFunc);
+   *
+   * The benefit of using this over `vjs.on(otherElement, 'eventName', myFunc)`
+   * and `otherComponent.on('eventName', myFunc)` is that this way the listeners
+   * will be automatically cleaned up when either component is disposed.
+   * It will also bind myComponent as the context of myFunc.
+   *
+   * **NOTE**: When using this on elements in the page other than window
+   * and document (both permanent), if you remove the element from the DOM
+   * you need to call `vjs.trigger(el, 'dispose')` on it to clean up
+   * references to it and allow the browser to garbage collect it.
+   *
+   * @param  {String|vjs.Component} first   The event type or other component
+   * @param  {Function|String}      second  The event handler or event type
+   * @param  {Function}             third   The event handler
+   * @return {vjs.Component}        self
+   */
+  on(first, second, third){
+    var target, type, fn, removeOnDispose, cleanRemover, thisComponent;
```

Comment:
Also, it would be nice to move all this `on` stuff into the `EventEmitter` class we have and then have Component extend from it or something like that.